ledger/ledger_test.go

```diff
@@ -1567,6 +1650,141 @@ func TestListAssetsAndApplications(t *testing.T) {
 	require.Equal(t, appCount, len(results))
 }
 
+// TestLedgerKeepsOldBlocksForStateProof test that if stateproof chain is delayed for X intervals,  the ledger will not
```

Comment:
Since TxTail exists, if StateProofMaxRecoveryIntervals is less than MaxTxnLifeTime, it's nicer to use it, since then we can remove logic to hold onto entire blocks when we only need the header.

Reply:
we can debate on what is the exact value in `StateProofMaxRecoveryIntervals` but currently, it is 10, which means that if state proof is lagging, the ledger needs to retain ~2560 blocks back (which is more than `MaxTxnLife`). It is possible to implement the same extension logic (to keep more block headers than 1000 if needed) and remove it from the block queue. but it is not a small task.